src/main/java/de/tum/in/www1/artemis/web/rest/PlagiarismResource.java

```diff
@@ -1,46 +1,238 @@
+
+        // check if current user is part of the comparison or not
+        if (!(comparison.getSubmissionA().getStudentLogin().equals(user.getLogin()) || comparison.getSubmissionB().getStudentLogin().equals(user.getLogin()))) {
+            throw new AccessForbiddenException("User tried updating plagiarism case they're not affected by.");
+        }
+
+        PlagiarismComparison anonymizedComparisonForStudentView = this.plagiarismService.anonymizeComparisonForStudentView(comparison, user.getLogin());
+        return ResponseEntity.ok(new PlagiarismCaseDTO(anonymizedComparisonForStudentView.getPlagiarismResult().getExercise(), Set.of(anonymizedComparisonForStudentView)));
+    }
+
+    /**
+     * Updates a student statement on a plagiarismComparison.
+     * I.e. one of the students that is accused of plagiarising updates/sets the respective/individual response/defence
+     *
+     * @param comparisonId of the comparison
+     * @param statement the students statement
+     * @return the student statement
+     */
+    @PutMapping("plagiarism-comparisons/{comparisonId}/student-statement")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<PlagiarismStatementDTO> updatePlagiarismComparisonStudentStatement(@PathVariable("comparisonId") long comparisonId,
+            @RequestBody PlagiarismStatementDTO statement) {
+        var comparison = plagiarismComparisonRepository.findByIdElseThrow(comparisonId);
+        User user = userRepository.getUser();
+        String studentLogin = user.getLogin();
+        String studentStatement = statement.statement;
+
+        if (comparison.getInstructorStatementA() != null && comparison.getSubmissionA().getStudentLogin().equals(studentLogin)) {
+            plagiarismComparisonRepository.updatePlagiarismComparisonStudentStatementA(comparison.getId(), studentStatement);
+        }
+        else if (comparison.getInstructorStatementB() != null && comparison.getSubmissionB().getStudentLogin().equals(studentLogin)) {
+            plagiarismComparisonRepository.updatePlagiarismComparisonStudentStatementB(comparison.getId(), studentStatement);
+        }
+        else {
+            throw new AccessForbiddenException("User tried updating plagiarism case they're not affected by.");
+        }
+        return ResponseEntity.ok(statement);
+    }
+
+    /**
+     * Updates the final status of a plagiarism comparison concerning one of both students.
+     * This process will send a notification to the respective student.
+     * I.e. an instructor sends his final verdict/decision
+     *
+     * @param comparisonId of the comparison
+     * @param studentLogin of the student
+     * @param statusDTO is the final status of this plagiarism comparison concerning one of both students
+     * @return the final (updated) status of this plagiarism comparison concerning one of both students
+     */
+    @PutMapping("plagiarism-comparisons/{comparisonId}/{studentLogin}/final-status")
```

Comment:
does not follow our REST URL best practices, please improve it:
- see comment above

Reply:
All endpoints adapted in fe01c3c